libsolidity/interface/SourceReferenceFormatter.cpp

```diff
@@ -49,6 +49,21 @@ void SourceReferenceFormatter::printSourceLocation(
 	if (startLine == endLine)
 	{
 		string line = scanner.lineAtPosition(_location->start);
+
+		int locationLength = endColumn - startColumn;
+		if (locationLength > 150)
+		{
+			line = line.substr(0, startColumn + 75) + " ... " + line.substr(endColumn);
```

Comment:
Why did you remove the context before `endColumn`?

Reply:
As mentioned in the previous commit comment , i suggest if the source location is > 150 chars, we only take the first 75 chars and append " ... " as a shortened source location.

Reply:
Sure, but sometimes the end might be more important than the beginning.

Reply:
So any suggestions on how to shorten the source location?

Reply:
keep x characters after start and x characters before end so that the whole `line` string has a length of 150.

Reply:
ah no, sorry. Something like
`line = line.substr(0, startColumn + 70) + " ... " + line.substr(endColumn - 70);`

Reply:
Sorry about the back-and-forth here.

Reply:
I understand that, but if I take the first 70 and last 70 chars of source location and the dots, it will be 145 chars for source location. If there is code around the source location, most likely the total length of line will be > 150, and the code around the source location will be converted to " ... ". Is that fine?

Reply:
Yeah, I see. So what we probably want is some chars left and right of the beginning and end of the source location. So what about 
`line.substr(0, startColumn + 35) + " ... " + line.substr(endColumn - 35)` ?

Reply:
yes that will be good